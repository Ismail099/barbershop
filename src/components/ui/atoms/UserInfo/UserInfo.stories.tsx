import { Story, Meta } from '@storybook/react';

import { UserInfo, UserInfoProps } from './UserInfo';

export default {
  title: 'atoms/UserInfo',
  component: UserInfo,
  args: {
    userImage: '/assets/img/testUser.png',
    userName: 'gabriel guedes'
  }
} as Meta;

export const Basic: Story<UserInfoProps> = (args) => <UserInfo {...args} />;
