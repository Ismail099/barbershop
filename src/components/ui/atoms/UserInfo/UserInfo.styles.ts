import styled, { css } from 'styled-components';
import { UserInfoProps } from './UserInfo';

export const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
`;
export const User = styled.div<Pick<UserInfoProps, 'userImage'>>`
  ${({ theme, userImage }) => css`
    position: relative;
    display: flex;
    align-items: center;
    justify-content: center;
    width: 4.5rem;
    height: 4.5rem;
    border-radius: 50%;
    background-image: url(${userImage});
    background-size: cover;

    &::before {
      content: '';
      position: absolute;
      width: calc(100% + 0.5rem);
      height: calc(100% + 0.5rem);
      border-radius: 50%;
      border: 0.2rem solid ${theme.colors.primary};
    }
  `}
`;
export const Name = styled.p`
  ${({ theme }) => css`
    text-transform: capitalize;
    font-size: ${theme.font.sizes.xsmall};
    font-weight: ${theme.font.thin};
    color: ${theme.colors.textName};
    margin-top: 1rem;
  `}
`;
