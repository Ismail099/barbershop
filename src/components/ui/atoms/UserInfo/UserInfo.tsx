import * as S from './UserInfo.styles';

export type UserInfoProps = {
  userImage: string;
  userName: string;
};

export const UserInfo = ({ userImage, userName }: UserInfoProps) => (
  <S.Wrapper>
    <S.User userImage={userImage} aria-label="User image" />
    <S.Name aria-label="User name">{`-${userName}`}</S.Name>
  </S.Wrapper>
);
