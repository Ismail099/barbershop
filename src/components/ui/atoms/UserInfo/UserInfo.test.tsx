import { screen } from '@testing-library/react';
import { renderWithTheme } from '~src/utils/tests/helpers';

import { UserInfo } from './UserInfo';

const props = {
  userImage: '/assets/img/testUser.png',
  userName: 'Gabriel Guedes'
};

describe('<UserInfo />', () => {
  it('Deve renderizar o componente <UserInfo />', () => {
    //Arrange
    const { container } = renderWithTheme(<UserInfo {...props} />);
    //Assert
    expect(screen.getByText(`-${props.userName}`)).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });
});
