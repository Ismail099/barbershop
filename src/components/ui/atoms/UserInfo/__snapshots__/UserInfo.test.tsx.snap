// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserInfo /> Deve renderizar o componente <UserInfo /> 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 4.5rem;
  height: 4.5rem;
  border-radius: 50%;
  background-image: url(/assets/img/testUser.png);
  background-size: cover;
}

.c1::before {
  content: '';
  position: absolute;
  width: calc(100% + 0.5rem);
  height: calc(100% + 0.5rem);
  border-radius: 50%;
  border: 0.2rem solid #1affea;
}

.c2 {
  text-transform: capitalize;
  font-size: 1.2rem;
  font-weight: 300;
  color: #A5A5A5;
  margin-top: 1rem;
}

<div
  class="c0"
>
  <div
    aria-label="User image"
    class="c1"
  />
  <p
    aria-label="User name"
    class="c2"
  >
    -Gabriel Guedes
  </p>
</div>
`;
